core_impl/src/test/java/com/google/instrumentation/trace/TraceExporterImplTest.java

```diff
@@ -149,30 +114,72 @@ public void serviceHandlerThrowsException() {
         .when(mockServiceHandler)
         .export(anyListOf(SpanData.class));
     traceExporter.registerServiceHandler("mock.service", mockServiceHandler);
-    traceExporter.addSpan(generateSpan(SPAN_NAME_1));
+    SpanImpl span1 = createEndedSpan(SPAN_NAME_1);
     List<SpanData> exported = serviceHandler.waitForExport(1);
     assertThat(exported.size()).isEqualTo(1);
-    assertThat(exported.get(0).getDisplayName()).isEqualTo(SPAN_NAME_1);
+    assertThat(exported.get(0)).isEqualTo(span1.toSpanData());
     // Continue to export after the exception was received.
-    traceExporter.addSpan(generateSpan(SPAN_NAME_1));
+    SpanImpl span2 = createEndedSpan(SPAN_NAME_1);
     exported = serviceHandler.waitForExport(1);
     assertThat(exported.size()).isEqualTo(1);
-    assertThat(exported.get(0).getDisplayName()).isEqualTo(SPAN_NAME_1);
+    assertThat(exported.get(0)).isEqualTo(span2.toSpanData());
   }
 
   @Test
   public void exportSpansToMultipleServices() {
     FakeServiceHandler serviceHandler2 = new FakeServiceHandler();
     traceExporter.registerServiceHandler("test.service2", serviceHandler2);
-    traceExporter.addSpan(generateSpan(SPAN_NAME_1));
-    traceExporter.addSpan(generateSpan(SPAN_NAME_2));
+    SpanImpl span1 = createEndedSpan(SPAN_NAME_1);
+    SpanImpl span2 = createEndedSpan(SPAN_NAME_2);
     List<SpanData> exported1 = serviceHandler.waitForExport(2);
     List<SpanData> exported2 = serviceHandler2.waitForExport(2);
     assertThat(exported1.size()).isEqualTo(2);
     assertThat(exported2.size()).isEqualTo(2);
-    assertThat(exported1.get(0).getDisplayName()).isEqualTo(SPAN_NAME_1);
-    assertThat(exported2.get(0).getDisplayName()).isEqualTo(SPAN_NAME_1);
-    assertThat(exported1.get(1).getDisplayName()).isEqualTo(SPAN_NAME_2);
-    assertThat(exported2.get(1).getDisplayName()).isEqualTo(SPAN_NAME_2);
+    assertThat(exported1.get(0)).isEqualTo(span1.toSpanData());
+    assertThat(exported2.get(0)).isEqualTo(span1.toSpanData());
+    assertThat(exported1.get(1)).isEqualTo(span2.toSpanData());
+    assertThat(exported2.get(1)).isEqualTo(span2.toSpanData());
+  }
```

Comment:
I think there should be a test that checks that non-sampled spans are not exported.